### PR TITLE
SAK-48719 grader: more scroll fatigue fixes

### PR DESCRIPTION
**Pull request by adrianfish**
https://sakaiproject.atlassian.net/browse/SAK-48719

Also, based on more feedback from Wilma Hodges, the comment and private notes sections now display current entries in a truncated and graduated fashion. This way instructors can see at a glance that there is a comment, and they see the first snippet of that comment. I've also made the remove treatment of rubric, feedback and private notes similar. Each one has a remove link.

### CODE REVIEW

**Comment by adrianfish** [comment]
Screenshot 2023-04-21 at 10 22 49
Screenshot 2023-04-21 at 10 22 27
Screenshot 2023-04-21 at 10 21 41

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 1]
```diff
@@ -1323,47 +635,51 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
-  next() {
+  _next() {
+
     const currentIndex = this.submissions.findIndex(s => s.id === this.submission.id);
 
     if (currentIndex < this.submissions.length - 1) {
-      if (!this.canNavigate()) {
+      if (!this._canNavigate()) {
         return;
       }
```
Comment (kunaljaykam) [inline webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:671]:
```suggestion
      if (!this._canNavigate()) return;
```

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 2]
```diff
@@ -1323,47 +635,51 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
     }
   }
 
-  studentSelected(e) {
-    if (!this.canNavigate()) {
+  _studentSelected(e) {
+
+    if (!this._canNavigate()) {
       return;
     }
```
Comment (kunaljaykam) [inline webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:659]:
```suggestion
    if (!this._canNavigate()) return;
```

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 3]
```diff
@@ -1213,20 +521,22 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
   }
 
-  cancel() {
-    if (!this.canNavigate()) {
+  _cancel() {
+
+    if (!this._canNavigate()) {
       return;
     }
```
Comment (kunaljaykam) [inline webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:532]:
```suggestion
    if (!this._canNavigate()) return;
```

File: webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js [hunk 4]
```diff
@@ -240,652 +155,62 @@ export class SakaiGrader extends gradableDataMixin(SakaiElement) {
...
-  }
+  _checkOverflow(el) {
+
+    if (!el) return false;
+
+    const curOverflow = el.style.overflow;
+
+    if (!curOverflow || curOverflow === "visible") {
+      el.style.overflow = "hidden";
+    }
```
Comment (kunaljaykam) [inline webcomponents/tool/src/main/frontend/js/grader/sakai-grader.js:202]:
```suggestion
    if (!curOverflow || curOverflow === "visible") el.style.overflow = "hidden";
```

**Review by kunaljaykam — COMMENTED** [review]
Everything looks good @adrianfish, I have left some suggestions for you to take a look at. Great work!

**Pull request merged**